### PR TITLE
Alternative implementation: Persistent BMT data for blocks

### PR DESCRIPTION
**Pull request by xgreenx**
It is an alternative implementation of the https://github.com/FuelLabs/fuel-core/pull/913 with minimum changes. The main change is the same, but other changes are minimal.

### CODE REVIEW

**Comment by xgreenx** [comment]
@Voxelot I used `master` as the base because it shows the final diff(without overriding changes in the parent PR).
Plus, we don't need to merge this -> parent -> master=)

File: crates/fuel-core/src/database/storage.rs
Comment (Voxelot) [inline crates/fuel-core/src/database/storage.rs:73]: Is the idea to keep these mappables here instead of fuel-core-storage because they aren't needed anywhere else other than fuel-core right now?

File: crates/services/importer/src/importer/test.rs
Comment (Voxelot) [inline crates/services/importer/src/importer/test.rs:223]: test_case's should be documented with human readable descriptions, ie.

`#[test_case(...; "successfully imports genesis block when database is empty")`

Comment (Voxelot) [reply crates/services/importer/src/importer/test.rs:223]: created issue #940 to address this as a followup.

**Review by Voxelot — APPROVED** [review]

**Review by bvrooman — APPROVED** [review]
Reapproving after dismissing @Voxelot's approval

**Review by Voxelot — APPROVED** [review]

**Pull request merged**
